packages/kit/src/runtime/client/client.js

```diff
@@ -520,7 +520,17 @@ function get_navigation_result_from_branch({ url, params, branch, status, error,
 		props: {
 			// @ts-ignore Somehow it's getting SvelteComponent and SvelteComponentDev mixed up
 			constructors: compact(branch).map((branch_node) => branch_node.node.component),
-			page
+			// we need to assign a new page object so that subscribers are correctly notified
+			page: {
+				data: page.data,
+				error: page.error,
+				form: page.form,
+				params: page.params,
+				route: page.route,
+				state: page.state,
+				status: page.status,
+				url: page.url
```

Comment:
- is there a reason why we can't just do `page: { ...page }`?
- if we can't do that, then this should be moved into a helper function and reused across the change sites to reduce a bit of code size

Reply:
Yeah, no clue why but spreading with `{ ...page }` just returns an empty object. Maybe @Rich-Harris knows

Reply:
ah, it's because of the class with `$state.raw` on it, producing getters which are not spreadable